app/src/org/commcare/dalvik/dialogs/AlertDialogFragment.java

```diff
@@ -32,6 +34,11 @@ public void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
+    public void onCancel(DialogInterface dialog) {
+        factory.performCancel(dialog);
```

Comment:
Yeah, I guess its okay, just feels like an unnecessary level of indirection. The framework should be doing this work for us; I just wasn't able to find out how to get it to do that.